src/Microsoft.AspNet.Mvc.Razor/Precompilation/RazorErrorExtensions.cs

```diff
@@ -20,15 +22,19 @@ public static Diagnostic ToDiagnostics([NotNull] this RazorError error, [NotNull
                 defaultSeverity: DiagnosticSeverity.Error,
                 isEnabledByDefault: true);
 
-            var textSpan = new TextSpan(error.Location.AbsoluteIndex, error.Length);
-            var linePositionStart = new LinePosition(error.Location.LineIndex, error.Location.CharacterIndex);
-            var linePositionEnd = new LinePosition(error.Location.LineIndex,
-                                                   error.Location.CharacterIndex + error.Length);
-            var linePositionSpan = new LinePositionSpan(linePositionStart, linePositionEnd);
-
-            var location = Location.Create(filePath, textSpan, linePositionSpan);
+            var location = error.Location;
+            if (location == SourceLocation.Undefined)
+            {
+                location = SourceLocation.Zero;
+            }
+            var length = Math.Max(0, error.Length);
```

Comment:
I was talking about the case where `error.Length > 0` and `location == SourceLocation.Zero` (might have been `SourceLocation.Undefined` a few lines earlier).  I would prefer `length == 0` for this case, e.g.
``` c#
var location = error.Location;
if (location == SourceLocation.Undefined)
{
  location = SourceLocation.Zero;
}

var length = Math.Max(0, error.Length);
if (location == SourceLocation.Zero)
{
  length = 0;
}
```